### PR TITLE
Bump version to 0.10.0

### PR DESCRIPTION
**Pull request by kngwyu**
I think it's good timing to release.

### CODE REVIEW

**Comment by kngwyu** [comment]
#920 should be merged before the release.

File: CHANGELOG.md [hunk 1]
```diff
@@ -385,7 +387,8 @@ Yanked
 ### Added
 - Initial release
 
-[Unreleased]: https://github.com/pyo3/pyo3/compare/v0.9.2...HEAD
+[Unreleased]: https://github.com/pyo3/pyo3/compare/v0.10.0...HEAD
+[0.10.0]: https://github.com/pyo3/pyo3/compare/v0.10.0...HEAD
```
Comment (davidhewitt) [inline CHANGELOG.md:391]:
```suggestion
[0.10.0]: https://github.com/pyo3/pyo3/compare/v0.9.2...v0.10.0
```

File: guide/src/migration.md [hunk 2]
```diff
@@ -1,5 +1,37 @@
...
+After:
+```rust
+let gil = pyo3::Python::acquire_gil();
+let obj = gil.python().eval("lambda: 'Hi :)'", None, None).unwrap();
+let hi: &pyo3::types::PyString = obj.call0().unwrap().downcast().unwrap();
+assert_eq!(hi.len().unwrap(), 5);
+```
+
+### No `#![feature(specialization)]` in user code
+While PyO3 itself still requires spcialization and nightly Rust,
```
Comment (davidhewitt) [inline guide/src/migration.md:32]:
```suggestion
While PyO3 itself still requires specialization and nightly Rust,
```

File: guide/src/migration.md [hunk 3]
```diff
@@ -1,5 +1,37 @@
...
+
+After:
+```rust
+let gil = pyo3::Python::acquire_gil();
+let obj = gil.python().eval("lambda: 'Hi :)'", None, None).unwrap();
+let hi: &pyo3::types::PyString = obj.call0().unwrap().downcast().unwrap();
+assert_eq!(hi.len().unwrap(), 5);
+```
+
+### No `#![feature(specialization)]` in user code
```
Comment (davidhewitt) [inline guide/src/migration.md:31]: :tada:

File: guide/src/migration.md [hunk 4]
```diff
@@ -1,5 +1,37 @@
...
+
+This guide can help you upgrade code through breaking changes from one PyO3 version to the next.
+For a detailed list of all changes, see [CHANGELOG.md](https://github.com/PyO3/pyo3/blob/master/CHANGELOG.md)
+
+## from 0.9.* to 0.10
+
+### `ObjectProtocol` is removed
+All methods are moved to [`PyAny`].
+And since now all native types (e.g., `PyList`) implements `Deref<Target=PyAny>`,
+al you need is removing `ObjectProtocol` from your code.
```
Comment (davidhewitt) [inline guide/src/migration.md:11]:
```suggestion
all you need to do is remove `ObjectProtocol` from your code.
```

File: guide/src/migration.md [hunk 5]
```diff
@@ -1,5 +1,37 @@
 # Appendix B: Migrating from older PyO3 versions
-This guide can help you upgrade code through breaking changes from one PyO3 version to the next. For a detailed list of all changes, see [CHANGELOG.md](https://github.com/PyO3/pyo3/blob/master/CHANGELOG.md)
+
+This guide can help you upgrade code through breaking changes from one PyO3 version to the next.
+For a detailed list of all changes, see [CHANGELOG.md](https://github.com/PyO3/pyo3/blob/master/CHANGELOG.md)
+
+## from 0.9.* to 0.10
+
+### `ObjectProtocol` is removed
```
Comment (davidhewitt) [inline guide/src/migration.md:8]: Is it worth noting that `ObjectProtocol` was also in the prelude? So for many users this change may have been totally invisible.

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+
+## [0.10.0]
```
Comment (davidhewitt) [inline CHANGELOG.md:9]: I've been wondering about making a quick PR to make the changelog messages for `0.10.0` (hopefully) a little easier to read. Happy for me to do that?

Comment (kngwyu) [reply CHANGELOG.md:9]: > I've been wondering about making a quick PR to make the changelog messages for 0.10.0 (hopefully) a little easier to read.

OK :+1:  but it's easier for me if you specify what should be changed here.

Comment (kngwyu) [reply CHANGELOG.md:9]: Or maybe you can push the change directly to this branch.

Comment (davidhewitt) [reply CHANGELOG.md:9]: :+1: will push later

**Comment by davidhewitt** [comment]
I've pushed a bunch of rewording to the CHANGELOG for `0.10.0`. Added some missing stuff and tried to make everything more consistent / simpler to read.

I tried to draw inspiration from [Gitlab's guidelines](https://docs.gitlab.com/ee/development/changelog.html), which I think leads to a pretty good [changelog](https://gitlab.com/gitlab-org/gitlab-foss/blob/master/CHANGELOG.md).

Please edit anything you don't like about what I added / wrote.

**Comment by kngwyu** [comment]
:+1:

**Pull request merged**